packages/fiori-freestyle-writer/src/index.ts

```diff
@@ -133,21 +151,38 @@ async function generate<T>(basePath: string, data: FreestyleApp<T>, fs?: Editor)
         JSON.parse(render(fs.read(join(tmplPath, 'common', 'extend', 'package.json')), ffApp, {}))
     );
 
+    const addTests = ffApp.appOptions?.addTests;
     const packageJson: Package = JSON.parse(fs.read(packagePath));
+
     if (isEdmxProjectType) {
+        const addMock = !!ffApp.service?.metadata;
         // Add scripts for non-CAP applications
         packageJson.scripts = {
             ...packageJson.scripts,
             ...getPackageScripts({
                 localOnly: !!ffApp.service && !ffApp.service?.url,
-                addMock: !!ffApp.service?.metadata,
+                addMock,
                 sapClient: ffApp.service?.client,
                 flpAppId: ffApp.app.flpAppId,
                 startFile: data?.app?.startFile,
                 localStartFile: data?.app?.localStartFile,
-                generateIndex: ffApp.appOptions?.generateIndex
+                generateIndex: ffApp.appOptions?.generateIndex,
+                addTest: addTests
             })
         };
+        if (addTests) {
+            addTestScripts(packageJson, addMock);
+            const config = {
+                appId: ffApp.app.id,
+                applicationDescription: ffApp.app.description,
+                applicationTitle: ffApp.app.title,
+                viewName: (ffApp.template.settings as BasicAppSettings).viewName,
+                ui5Theme: ffApp.ui5?.ui5Theme,
+                ui5Version: ffApp.ui5?.version,
+                enableTypeScript: ffApp.appOptions?.typescript
+            };
+            await generateFreestyleOPAFiles(basePath, config, fs, log);
```

Comment:
Why not put all the test related code (in the condition) into a single function:
```
if (addTests) {
    generateOPATests()
}
```